src/main/java/net/digibi/mailerlite/MailerLiteClient.java

```diff
@@ -12,34 +15,92 @@
-        HttpResponse<String> response = Unirest
-            .post(API_END_POINT + "groups/" + groupId + "/subscribers")
-            .header("content-type", "application/json")
-            .header("x-mailerlite-apikey", API_KEY)
-            .body(userBody)
-            .asString();
+    /**
+     * Moves user provided in userBody to specified groupId
+     * @param groupId
+     * @param userBody
+     * @return
+     */
+    public boolean moveSubscriberToGroup(long groupId, String userBody) {
+        HttpResponse<String> response = Unirest.post(API_END_POINT + "groups/" + groupId + "/subscribers").header("content-type", "application/json").header("x-mailerlite-apikey", API_KEY)
+                .body(userBody).asString();
         return response.isSuccess();
     }
 
-    public boolean subscribeUser(long groupId, String email, String name) {
-        return subscribeUser(groupId, createUserBodyToSend(email, name, null));
+    /**
+     * Moves user(identified by email, name) to specified groupId and set type of subscription
+     * @param groupId
+     * @param email
+     * @param type allowed values: unsubscribed, active, unconfirmed
+     * @return
+     */
+    public boolean moveSubscriberToGroup(long groupId, String email, String type) {
+        return moveSubscriberToGroup(groupId, createUserBodyToSend(email, null, type));
     }
 
-    public boolean subscribeUser(long groupId, String email, String name, String fields) {
-        return subscribeUser(groupId, createUserBodyToSend(email, name, fields));
+    /**
+     * Moves user(identified by email, name) to specified groupId, set fields & type of subscription
+     * @param groupId
+     * @param email
+     * @param fields fields to set
+     * @param type allowed values: unsubscribed, active, unconfirmed
+     * @return
+     */
+    public boolean moveSubscriberToGroup(long groupId, String email, Map<String, String> fields, String type) {
+        return moveSubscriberToGroup(groupId, createUserBodyToSend(email, fields, type));
     }
 
-    public static String createUserBodyToSend(String email, String name, String fields) {
-        JSONObject jsonObject = new JSONObject().put("email", email).put("name", name);
-        if (fields!=null) {
-            jsonObject.put("fields", fields);
+    /**
+     * Method to construct String in JSON format
+     * 
+     * @param email
+     * @param fields
+     * @param type   values accepted: unsubscribed, active
+     * @return String in JSON format
+     */
+    public static String createUserBodyToSend(String email, Map<String, String> fields, String type) {
+        JSONObject jsonObject = new JSONObject().put("email", email).put("type", type);
+        if (fields != null) {
+            jsonObject.put("fields", fields.toString());
         }
         return jsonObject.toString();
     }
+    
+    public HttpResponse<String> updateUserDetails(int identifier, Map<String, String> fields) {
```

Comment:
MailerLite support two update types:
> type string
> Available values: unsubscribed, active

If you want to use `updateUserDetails` as method name, add another parameter to take an enum of update type, and add it to the body. Or you can separate the method into two - one for each type. I prefer the first method as they have the same API endpoint, but I'm open to other suggestions as well.